pkgs/applications/science/math/sage/sage.nix

```diff
@@ -58,7 +49,7 @@ stdenv.mkDerivation rec {
     quicktest = sage-tests.override { longTests = false; timeLimit = 600; }; # as many tests as possible in ~10m
     doc = sagedoc;
     lib = sage-with-env.env.lib;
-    kernelspec = jupyter-kernel-definition;
+    kernelspec = jupyter-kernel-specs.definition.sagemath;
```

Comment:
Hi, thanks for updating. The function [`jupyter-kernel.create`](https://github.com/NixOS/nixpkgs/blob/e69af2535c82ab9b85eea6dca1ceb435fee7bb2d/pkgs/applications/editors/jupyter/kernel.nix#L29) does not re-expose the definitions passed to it and hence causes the error

```
error: attribute 'definition' missing

       at <truncated path>/nixpkgs/pkgs/applications/science/math/sage/sage.nix:52:18:

           51|     lib = sage-with-env.env.lib;
           52|     kernelspec = jupyter-kernel-specs.definition.sagemath;
             |                  ^
           53|   };

```